web/App.js

```diff
@@ -0,0 +1,235 @@
+                const assetKey = String.fromCharCode.apply(null, assetKeyArray.filter(x => x));
+                
+                if (gameAssets[assetKey]['type'] === 'audio') {
+                    if (audioCtx) {
+                        source = audioCtx.createBufferSource();
+                        source.connect(audioCtx.destination);
+                        source.buffer = gameAssets[assetKey].data;
+                        source.start(0);
+                    } else {
+                        console.warn("Cant play audio");
+                    }
+                } else {
+                
+                    let image;
+                    if (imageCache[assetKey]) {
+                        image = imageCache[assetKey];
+                        ctx.drawImage(image, (assetPosX / 100) * horizontalScale, 
+                            (assetPosY / 100) * verticalScale, image.width, image.height)
+
+                    } else {
+                        image = new Image(assetSizeX / 100 * horizontalScale, assetSizeY / 100 * verticalScale);
+                        imageCache[assetKey] = image;
+                        image.onload = () => {
+                            ctx.drawImage(image, (assetPosX / 100) * horizontalScale, 
+                                (assetPosY / 100) * verticalScale, image.width, image.height)
+                        };
+
+                        image.src = gameAssets[assetKey].data;
+                    }
+                }
+            }
+
+            i += frameSize;
+
+        }
+    }
+
+};
+
+const click = function(x, y) {
+    const pixelWidth = canvas.width / originalWidth;
+    const pixelHeight = canvas.height / originalHeight;
+    const clickX = Math.floor(x / pixelWidth);
+    const clickY = Math.floor(y  / pixelHeight);
+    const payload = {type: 'click',  data: {x: clickX, y: clickY}};
+    socket.send(JSON.stringify(payload));
+};
+
+const keydown = function(key) {
+    const payload = {type: 'keydown',  key: key};
+    socket.send(JSON.stringify(payload));
+};
+
+const keyup = function(key) {
+    const payload = {type: 'keyup',  key: key};
+    socket.send(JSON.stringify(payload));
+};
+
+const unlock = () => {
+    if (!audioCtx) {
+        audioCtx = new (window.AudioContext || window.webkitAudioContext)(); 
+        if (audioCtx.state === 'suspended') {
+            audioCtx.resume();
+        }
+        for (const key in gameAssets) {
+            if (gameAssets[key]['type'] === 'audio' && !gameAssets[key]['decoded']) {
+                audioCtx.decodeAudioData(gameAssets[key].data, (buffer) => {
+                    gameAssets[key].data = buffer;
+                    gameAssets[key].decoded = true;
+                });
+            }
+        }
+    }
+}
+
+document.addEventListener('touchstart', unlock, false);
+
+canvas.addEventListener('mousedown', function(e) {
+    mouseDown = true;
+    unlock();
+});
+
+canvas.addEventListener('mouseup', function(e) {
```

Comment:
Once this list of event listeners grows and starts to split up more based on device, we should abstract some of this stuff